src/java/org/jivesoftware/openfire/muc/spi/LocalMUCRole.java

```diff
@@ -162,16 +162,7 @@ public void setPresence(Presence newPresence) {
         }
         this.presence = newPresence;
         this.presence.setFrom(getRoleAddress());
-        if (extendedInformation != null) {
-            // Remove any previous extendedInformation, then re-add it.
-            Element mucUser = presence.getElement().element(QName.get("x", "http://jabber.org/protocol/muc#user"));
-            if (mucUser != null) {
-                // Remove any previous extendedInformation, then re-add it.
-                presence.getElement().remove(mucUser);
-            }
-            Element exi = extendedInformation.createCopy();

```

Comment:
Yes, I even did this before and it worked, but I was unsure about possible hidden side effects, although I couldn't see any.
But I thought the original author surely had its reason to create a copy every time.